apis/kueue/webhooks/workload_webhook.go

```diff
@@ -21,29 +21,71 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	metav1validation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	cgcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
 )
 
-type WorkloadWebhook struct{}
+// LimitRangeProvider - hepler interface able to privide LimitRanges
+type LimitRangeProvider interface {
+	// GetContainersLimitRanges - retrive all the container relevant limit range item in the provided
+	// namespace
+	GetContainersLimitRanges(ctx context.Context, namespace string) ([]corev1.LimitRangeItem, error)
+}
+
+type limitRangeProvider struct {
+	getter cgcorev1.LimitRangesGetter
+}
+
+func (w *limitRangeProvider) GetContainersLimitRanges(ctx context.Context, namespace string) ([]corev1.LimitRangeItem, error) {
+	list, err := w.getter.LimitRanges(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if len(list.Items) == 0 {
+		return nil, nil
+	}
+
+	ret := make([]corev1.LimitRangeItem, 0, len(list.Items))
+	for _, lr := range list.Items {
+		for _, lri := range lr.Spec.Limits {
+			if lri.Type == corev1.LimitTypeContainer {
```

Comment:
the controller-runtime client has indexing capabilities. Use them to avoid filtering at this level.
For example https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/core/indexer/indexer.go

Reply:
Indexing could help, but only for the first loop, and in practice , is the same namespace, is unlikely to have multiple LimitRanges.

With the change in  `getDefaultsFromLimitRanges` these loops are only executed once for every web-hook call.